scipy/stats/_morestats.py

```diff
@@ -3512,9 +3518,26 @@ def wilcoxon(x, y=None, zero_method="wilcox", correction=False,
 
     n_zero = np.sum(d == 0)
     if n_zero > 0 and mode == "exact":
-        mode = "approx"
-        warnings.warn("Exact p-value calculation does not work if there are "
-                      "zeros. Switching to normal approximation.")
+        
+        if zero_method == "zsplit":
+        
+            # ties are accepted but the number of zeros must be even
+            # if it is odd, one zero is discarded
+            if n_zero % 2 == 1:
+                # discard one zero difference
+                # first [0] is for the tuple returned by np.where, 
+                # second [0] is for the first element of the tuple
+                where_idx = np.where(d == 0)[0][0]
+                x = np.delete(x, where_idx)
+                if y is not None:
+                    y = np.delete(y, where_idx)
+                d = np.delete(d, where_idx)
+                n_zero -= 1
+                
+        else:
+            mode = "approx"
+            warnings.warn("Exact p-value calculation does not work if there are "
+                        "zeros. Switching to normal approximation.")
 
     if mode == "approx":
         if zero_method in ["wilcox", "pratt"]:
```

Comment:
Probably for another PR, but it feels like all these `if`s became too complex.

Perhaps an exhaustive `if ... and ... and ...` with the values of `method`, `zero_method` would be easier to understand?

Reply:
It is getting complex, but let's not refactor that right now.